### PR TITLE
FUSETOOLS-1826

### PR DESCRIPTION
**Pull request by vcornejo**
publish a TOPIC_REMOVE_CAMEL_ELEMENT Event to clean markers

### CODE REVIEW

File: core/tests/org.fusesource.ide.camel.tests.util/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -6,6 +6,24 @@ Bundle-Version: 9.1.0.qualifier
...
- org.fusesource.ide.camel.model.service.v2_17_3;bundle-version="9.1.0"
-Export-Package: org.fusesource.ide.camel.tests.util
+ org.fusesource.ide.camel.model.service.v2_17_3;bundle-version="9.1.0",
+ org.fusesource.ide.camel.model.service.core.tests.integration,
+ org.fusesource.ide.branding,
+ org.eclipse.core.runtime;bundle-version="3.11.1",
+ org.eclipse.core.resources,
+ org.eclipse.core.databinding;bundle-version="1.5.0",
+ org.eclipse.jdt.ui,
+ org.eclipse.swt,
```
Comment (apupier) [inline core/tests/org.fusesource.ide.camel.tests.util/META-INF/MANIFEST.MF]: org.eclipse.swt is most of the time not a real required dependency and should be avoided

File: core/tests/org.fusesource.ide.camel.tests.util/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -6,6 +6,24 @@ Bundle-Version: 9.1.0.qualifier
...
+ org.eclipse.jface,
+ org.eclipse.pde.core,
+ org.eclipse.graphiti,
+ org.eclipse.emf.common;bundle-version="2.11.0",
+ org.eclipse.emf.transaction;bundle-version="1.8.0",
+ org.fusesource.ide.branding;bundle-version="9.1.0",
+ org.fusesource.ide.camel.editor;bundle-version="9.1.0",
+ org.eclipse.ui.ide;bundle-version="3.12.0",
+
```
Comment (apupier) [inline core/tests/org.fusesource.ide.camel.tests.util/META-INF/MANIFEST.MF]: empty line

File: core/tests/org.fusesource.ide.camel.tests.util/src/org/fusesource/ide/camel/test/util/editor/AbstractCamelEditorIT.java [hunk 3]
```diff
@@ -81,7 +81,7 @@ public void tearDown() throws Exception {
 	}
 
 	protected IEditorPart openFileInEditor(String filePath) throws Exception {
-		InputStream inputStream = CamelEditorIT.class.getClassLoader().getResourceAsStream(filePath);
+		InputStream inputStream = AbstractCamelEditorIT.class.getClassLoader().getResourceAsStream(filePath);
```
Comment (apupier) [inline core/tests/org.fusesource.ide.camel.tests.util/src/org/fusesource/ide/camel/test/util/editor/AbstractCamelEditorIT.java]: I think that it should be getClass().getClassLoader()

File: editor/tests/org.fusesource.ide.camel.editor.tests.integration/META-INF/MANIFEST.MF [hunk 4]
```diff
@@ -28,6 +28,8 @@ Require-Bundle: org.junit;bundle-version="4.12.0",
  org.fusesource.ide.camel.model.service.core.tests.integration,
  org.fusesource.ide.branding;bundle-version="9.1.0",
  org.fusesource.ide.projecttemplates;bundle-version="9.1.0",
- org.eclipse.draw2d
+ org.eclipse.draw2d,
+ org.fusesource.ide.camel.tests.util
+Export-Package: org.fusesource.ide.camel.editor.integration
```
Comment (apupier) [inline editor/tests/org.fusesource.ide.camel.editor.tests.integration/META-INF/MANIFEST.MF]: why do you need to export package?

File: core/tests/org.fusesource.ide.camel.tests.util/META-INF/MANIFEST.MF (hunk 2 above)
Comment (vcornejo) [inline core/tests/org.fusesource.ide.camel.tests.util/META-INF/MANIFEST.MF]: ok

File: core/tests/org.fusesource.ide.camel.tests.util/src/org/fusesource/ide/camel/test/util/editor/AbstractCamelEditorIT.java (hunk 3 above)
Comment (vcornejo) [inline core/tests/org.fusesource.ide.camel.tests.util/src/org/fusesource/ide/camel/test/util/editor/AbstractCamelEditorIT.java]: you'r right

File: editor/tests/org.fusesource.ide.camel.editor.tests.integration/META-INF/MANIFEST.MF (hunk 4 above)
Comment (vcornejo) [inline editor/tests/org.fusesource.ide.camel.editor.tests.integration/META-INF/MANIFEST.MF]: Right, this was need only when AbstractCamelEditorIT is in editor.test.integration plugin. I will remove this Export.

**Comment by apupier** [comment]
testPR

**Comment by apupier** [comment]
build failure:

`04:39:19 [ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:0.26.0:compile (default-compile) on project org.fusesource.ide.camel.tests.util: Compilation failure: Compilation failure:
04:39:19 [ERROR] /home/jenkins/workspace/Fuse-Tooling-pullrequest/core/tests/org.fusesource.ide.camel.tests.util/src/org/fusesource/ide/camel/test/util/editor/AbstractCamelEditorIT.java:[116]
04:39:19 [ERROR] TransactionalEditingDomain editingDomain = CamelUtils.getDiagramEditor().getEditingDomain();
04:39:19 [ERROR] ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
04:39:19 [ERROR] Access restriction: The method 'DiagramEditor.getEditingDomain()' is not API (restriction on classpath entry '/home/jenkins/.m2/repository/p2/osgi/bundle/org.eclipse.graphiti.ui/0.13.0.v20160608-1043/org.eclipse.graphiti.ui-0.13.0.v20160608-1043.jar')
04:39:19 [ERROR] 1 problem (1 error)`

File: core/tests/org.fusesource.ide.camel.tests.util/META-INF/MANIFEST.MF [hunk 5]
```diff
@@ -6,6 +6,23 @@ Bundle-Version: 9.1.0.qualifier
...
+ org.eclipse.swt,
+ org.eclipse.jface,
+ org.eclipse.pde.core,
+ org.eclipse.graphiti,
+ org.eclipse.emf.common;bundle-version="2.11.0",
+ org.eclipse.emf.transaction;bundle-version="1.8.0",
+ org.fusesource.ide.branding;bundle-version="9.1.0",
+ org.fusesource.ide.camel.editor;bundle-version="9.1.0",
+ org.eclipse.ui.ide;bundle-version="3.12.0", 
+ org.fusesource.ide.camel.editor
```
Comment (apupier) [inline core/tests/org.fusesource.ide.camel.tests.util/META-INF/MANIFEST.MF]: I think that you need to add org.eclipse.graphiti.ui to require-bundle

**Comment by apupier** [comment]
testPR

File: core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidatorIT.java [hunk 6]
```diff
@@ -89,7 +89,7 @@ public void checkMarkers() throws Exception {
 
 		int initial = markers.length;
 		// there are errors or warnings
-		assertThat(initial > 0);
```
Comment (apupier) [inline core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidatorIT.java]: Optional:
you can write it in an even readable way:
assertThat(markers).hasSize(2);

File: core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidatorIT.java [hunk 7]
```diff
@@ -110,7 +110,7 @@ public void checkMarkers() throws Exception {
 		readAndDispatch(20);
 
 		markers = (IMarker[]) getAllMarkersMethod.invoke(problemView, new Object[] {});
-		assertThat(markers.length == 0);
```
Comment (apupier) [inline core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidatorIT.java]: OPTIONAL: assertThat(markers).isEmpty();

File: core/tests/org.fusesource.ide.camel.tests.util/META-INF/MANIFEST.MF (hunk 1 above)
Comment (apupier) [inline core/tests/org.fusesource.ide.camel.tests.util/META-INF/MANIFEST.MF]: Please remove it, if there are SWTException I'm pretty sure that there is/was other issues

File: core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidatorIT.java (hunk 6 above)
Comment (vcornejo) [inline core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidatorIT.java]: Yes, that's looks more readable.

File: core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidatorIT.java (hunk 7 above)
Comment (vcornejo) [inline core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidatorIT.java]: Yes, that's looks more readable.

**Comment by apupier** [comment]
Thank you Victor!
I really appreciate your effort to solve the issue and to conform to the Fuse Tooling standard.

**Pull request merged**